pkg/app/boot.go

```diff
@@ -1,24 +1,64 @@
 package app
 
 import (
+	"io"
+	"io/fs"
 	"os"
 	"path/filepath"
 
 	"github.com/pkg/errors"
 )
 
 // bootedPath is the path where to store the booted flag
-var bootedPath = "/var/run/modules"
+const bootedPath = "/var/run/modules"
+
+// bootManager contains the path of the booted files and fileSystem
+type bootManager struct {
+	bootedPath string
+	fs         fileSystem
+}
+
+type fileSystem interface {
+	Create(string) (io.ReadCloser, error)
+	MkdirAll(string, uint32) error
+	Stat(string) (any, error)
+}
+
+type defaultBootManager struct{}
+
+func (fs defaultBootManager) Create(path string) (io.ReadCloser, error) {
+	return os.Create(path)
+}
+
+func (dfs defaultBootManager) MkdirAll(path string, perm uint32) error {
+	return os.MkdirAll(path, fs.FileMode(perm))
+}
+
+func (dfs defaultBootManager) Stat(path string) (any, error) {
+	return os.Stat(path)
+}
+
+func defaultMarkManager() *bootManager {
```

Comment:
this complicates this code unecessary imho. You don't need a "manager" and then have to create an instance of it every time.

Instead do as per other comments